tests/test_abstract_model.py

```diff
@@ -247,10 +247,45 @@ def test_write_and_read(store_calc):
     # expected differences - n.b. order of calls above
     assert abcd_data_after_read["modified"] > abcd_data["modified"]
     assert abcd_data_after_read["uploaded"] > abcd_data["uploaded"]
-    assert abcd_data_after_read["hash"] != abcd_data["hash"]
 
     # expect results to match within fp precision
     for key in set(abcd_data.results_keys):
         assert abcd_data[key] == approx(
             np.array(abcd_data_after_read[key])
         ), f"{key}'s value does not match"
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        1296,
+        3.14,
+        [1, 2, 3],
+        (4, 5, 6),
+        {"a": "value"},
+        datetime.datetime.now(datetime.timezone.utc),
+        b"test",
+    ],
+)
+def test_hasher(data):
+    """Test hash calculated correctly."""
+    hasher_1 = Hasher()
+
+    # Test hash updated
+    init_hash = hasher_1()
+    hasher_1.update("Test value")
+    updated_hash = hasher_1()
+    assert updated_hash != init_hash
+
+    # Test updating hash for different data types
+    hasher_1.update(data)
+    assert updated_hash != hasher_1()
+
+    # Test newer hasher reset correctly
+    hasher_2 = Hasher()
+    assert hasher_2() == init_hash
+
+    # Test hashes match after same data added
+    hasher_2.update("Test value")
+    hasher_2.update(data)
+    assert hasher_1() == hasher_2()
```

Comment:
This seems like 4 different test cases, and the parametrization is useful for 2 of them. I suggest splitting this up into a test case for each of the 4 things you are testing, then we will have more granular view if things break as well.

Reply:
Good point, thanks! Should be done.